modules/xray/src/test/scala/XRayEnvironmentSuite.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019-2020 by Rob Norris and Contributors
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez
+package xray
+
+import cats.effect.std.Env
+import cats.effect.{IO, Resource}
+import cats.syntax.all.*
+import munit.{CatsEffectSuite, ScalaCheckEffectSuite}
+import natchez.xray.XRayEnvironmentSuite.TraceId
+import org.scalacheck.effect.PropF.forAllNoShrinkF
+import org.scalacheck.{Arbitrary, Gen}
+
+import scala.collection.immutable
+
+class XRayEnvironmentSuite extends CatsEffectSuite with ScalaCheckEffectSuite {
+
+  test("when using a Sync[F], get the Trace ID from one of many possible sources") {
+    forAllNoShrinkF { (maybeSystemProperty: Option[TraceId], maybeEnvVar: Option[TraceId]) =>
+      implicit val env: Env[IO] = new Env[IO] {
+        override def get(name: String): IO[Option[String]] =
+          entries.map(_.toMap.get(name))
+
+        override def entries: IO[immutable.Iterable[(String, String)]] =
+          maybeEnvVar
+            .map(_.value)
+            .toList
+            .tupleLeft("_X_AMZN_TRACE_ID")
+            .pure[IO]
+      }
+
+      maybeSystemProperty
```

Comment:
@armanbilge is there a nicer pattern for this, where I want to initialize and cleanup something if the option is set, and either way, run the `use` block?